app/domain/etl/ga/concerns/transform_path.rb

```diff
@@ -20,13 +20,10 @@ def transform_event_attributes(event)
     attributes = { page_path: sanitised_page_path }
 
     if duplicate_event
-      attributes[:pviews] = event.pviews + duplicate_event.pviews
-      attributes[:upviews] = event.upviews + duplicate_event.upviews
-      attributes[:useful_no] = event.useful_no + duplicate_event.useful_no
-      attributes[:useful_yes] = event.useful_yes + duplicate_event.useful_yes
-      attributes[:searches] = event.searches + duplicate_event.searches
-      attributes[:entrances] = event.entrances + duplicate_event.entrances
-      attributes[:exits] = event.exits + duplicate_event.exits
+      metric_names = (Metric.custom_metrics + Metric.ga_metrics).map(&:name)
+      metric_names.each do |metric|
+        attributes[metric.to_sym] = event.send(metric) + duplicate_event.send(metric) if event.respond_to?(metric)
```

Comment:
What does it mean is happening if `event.respond_to?(metric)` is false?

Reply:
I know it's super unlikely, but I'd like to guard against `NoMethodError` here.

Reply:
Wouldn't we want to know about an `NoMethodError` here, if it occurred? 

At the moment, attributes would just have a `nil` value I assume.

Reply:
Ah it's actually because one of our 'Custom' metrics (satisfaction) isn't an attribute of Events::GA and I can't think of an elegant way to handle that.

Reply:
So if there's an exception, it would be more readable to be explicit about it.

Maybe put `next if metric == 'satisfaction'` if the block, along with a comment, or maybe do:
```ruby
some_meaningful_variable = %w[satisfaction]
(metric_names - some_meaningful_variable).each do |metric|
 ....
end
```